ManagedIdentityTokenRevocation/Program.cs

```diff
@@ -0,0 +1,462 @@
+
+            while (elapsed < totalMs)
+            {
+                Console.Write($"\r  {spinnerFrames[frame]} ");
+                frame = (frame + 1) % spinnerFrames.Length;
+                Thread.Sleep(SpinnerIntervalMs);
+                elapsed += SpinnerIntervalMs;
+            }
+            Console.Write("\r   \n");
+        }
+
+        /// <summary>
+        /// Acquire a Key Vault token using the shared _managedIdentityApp, 
+        /// then retrieve the secret, and return the token.
+        /// </summary>
+        private static async Task<string> AcquireSecretFromKeyVaultAsync()
+        {
+            var token = string.Empty;
+
+            try
+            {
+                var authResult = await s_managedIdentityApp
+                    .AcquireTokenForManagedIdentity("https://vault.azure.net/.default")
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                token = authResult.AccessToken;
+                Console.WriteLine("  \u2713 Token for Key Vault acquired successfully!\n");
+
+                // Create a regular HttpClient
+                using var httpClient = new HttpClient();
+                httpClient.DefaultRequestHeaders.Authorization =
+                    new AuthenticationHeaderValue("Bearer", token);
+
+                // Retrieve a secret from Key Vault
+                var keyVaultUri = "https://revoguardkeyvault.vault.azure.net/secrets/RevoGuardSecret?api-version=7.5";
+                var response = await httpClient.GetAsync(keyVaultUri).ConfigureAwait(false);
+                var content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+
+                if (!response.IsSuccessStatusCode)
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine($"Key Vault request failed: {response.StatusCode}\n{content}");
+                    Console.ResetColor();
+                }
+                else
+                {
+                    using var doc = JsonDocument.Parse(content);
+                    if (doc.RootElement.TryGetProperty("value", out JsonElement secretValue))
+                    {
+                        Console.ForegroundColor = ConsoleColor.Green;
+                        Console.WriteLine("  \u2713 Secret from Key Vault:");
+                        Console.ResetColor();
+                        Console.WriteLine($"    {secretValue.GetString()}");
+                    }
+                    else
+                    {
+                        Console.WriteLine("Could not find 'value' property in JSON response.");
+                        Console.WriteLine(content);
+                    }
+                }
+            }
+            catch (MsalServiceException e)
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine($"MSAL Service Exception: {e.ErrorCode}");
+                PrintExceptionChain(e);
```

Comment:
Doesn't Console.WriteLine(e) do the same?

Reply:
PrintExceptionChain, just prints the Message and the StackTrace